src/image/image-proc.ts

```diff
@@ -0,0 +1,154 @@
+import Jimp from 'jimp';
+import { stdCalc } from './utils';
+import * as tf from '@tensorflow/tfjs-core';
+
+/**
+ * Image class contains utility to handle image manipulation
+ * @contructor (data: Jimp)
+ */
+export default class Image {
+  private img: Jimp; //img accessible to all methods in the class
+
+  constructor (data: Jimp) {
+    this.img = data;
+  }
+
+  /**
+   * Read image from files and create an Image object for processing
+   * @param name image file name
+   */
+  static async read(name: string): Promise<Image> {
+
+    const jimpObj = await Jimp.read(name);
+
+    return new Image(jimpObj);
+  }
+
+  /**
+   * Convert image Buffer to Tensor
+   * @param depth -> number of channel to obtain from image
+   * @return tf.Tensor3D
+   */
+  public toTensor(depth = 3): tf.Tensor3D {
+    return tf.browser.fromPixels({
+      data: this.data,
+      height: this.height,
+      width: this.width
+    }, depth);
+
+  }
+
+  /**
+   * Convert Image Tensor to one dimensional Array to be converted to Buffer.
+   * @param tensor Tensor3D of image
+   * @returns Promise<Image>
+   */
+  static async fromTensor(tensor: tf.Tensor3D): Promise<Image> {
+    const imgArray = await tf.browser.toPixels(tensor);
+    const data: Buffer = Buffer.from(imgArray);
+    const [ width, height ] = tensor.shape.slice(0, 2);
+    const img: Jimp = await new Jimp({ data:data, width:width, height:height });
+    return new Image(img);
+  }
+
+  /**
+   * Resize image
+   * @param width number
+   * @param height number
+   * @return Image
+   */
+  public resize(width:number, height:number): Image {
+    this.img = this.img.resize(width, height);
+
+    return this;
+  }
+
+  get width(): number {
+    return this.img.bitmap.width;
+  }
+
+  get height(): number {
+    return this.img.bitmap.height;
+  }
+
+  get data(): Buffer {
+    return this.img.bitmap.data;
+  }
+
+  /**
+   * save the image into a file
+   * @param name the image file name
+   * @return Boolean
+   */
+  public save(name:string): boolean {
+
+    const isSave = this.img.write(name);
+    if (isSave){
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Rotate image
+   * @param deg rotation degree
+   * @return Image
+   */
+  public rotate(deg:number): Image {
+    this.img = this.img.rotate(deg);
+    return this;
+  }
+
+  /**
+   * flip the image to horizontal or vertical
+   * @param horz
+   * @param vert
+   * @return Image
+   */
+  public flip(horz: boolean, vert: boolean): Image {
+    this.img = this.img.flip(horz, vert);
+    return this;
+  }
+
+  /**
+   * Crop the desired side of an image
+   * @param x
+   * @param y
+   * @param width
+   * @param height
+   * @return Image
+   */
+  public crop(x:number, y:number, width:number, height:number): Image {
+    this.img = this.img.crop(x, y, width, height);
+    return this;
+  }
+
+  /**
+   * Normalize images to [-1,1] using (tensor - mean) / std
```

Comment:
This function should be `standardization` and the range should be (-1, 1), IMO(Correct me if I am wrong).

For more details, we can refer to [this](https://en.wikipedia.org/wiki/Feature_scaling). (This really brings me back to stats class in college 💯 )

Reply:
Gentle ping @steveoni